internal/connector/test/integration/feature_test.go

```diff
@@ -116,17 +116,15 @@ func TestFeatures(t *testing.T) {
 			testName = strings.ReplaceAll(testName, "-", "_")
 
 			t.Run(testName, func(t *testing.T) {
-				// To preserve the current behavior, the test are market to be "safely" run in parallel, however
-				// we may think to introduce a new naming convention i.e. files that ends with _parallel would
-				// cause t.Parallel() to be invoked, other tests won't, so they won't be executed concurrently.
-				//
-				// This could help reducing/removing the need of explicit lock
-				t.Parallel()
+				// The assumption is that features run sequentially and scenario in a feature runs concurrently
```

Comment:
Great. I had issues with `Processor` integration tests and had to put a lock [around](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1686/files#diff-b43cae948ddaaa3a1207cf6d9eafb8f103309c0749568f7878dda0c9582f3dd1R75-R80) the whole scenario.